packages/melody-idom/__tests__/PatchInnerSpec.ts

```diff
@@ -97,6 +97,21 @@ describe("patching an element's children", () => {
 
                 expect(node).to.equal(div);
             });
+
+            describe('when Node.prototype.contains is not available', () => {
+                beforeEach(() => {
+                    container.contains = undefined;
```

Comment:
The `beforeEach` isn't needed. You only have a single test case.
Why introduce an additional level of nesting for the tests? ("patching an element's children" > "with an existing document tree" > "should return DOM node" > "when Node.prototype.contains is not available" > it "should return node from elementOpen")
There's some duplication, maybe your `describe` could be an `it`.

Reply:
The new level introduces a context where `Node.prototype.contains` it's not available, but if you think this is too verbose, we could have a one level `it` like:

```javascript
            it('should return node from elementOpen (Node.prototype.contains not available)', () => {
                container.contains = undefined;

                patchInner(container, () => {
                    node = elementOpen('div');
                    elementClose('div');
                });

                expect(node).to.equal(div);
            });
```
or similar